crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -190,8 +190,11 @@ impl<'db> SemanticIndexBuilder<'db> {
             countme::Count::default(),
         );
 
-        self.definitions_by_node
+        let existing_definition = self
+            .definitions_by_node
             .insert(definition_node.key(), definition);
```

Comment:
There are a few (or just one?) places where we call `self.definitions_by_node.extend`. Should we add the assertion there as well?

Reply:
Good point. There's one more place:

https://github.com/astral-sh/ruff/blob/d88c0cc0ebe1e149d43c600d07a512662b137e71/crates/red_knot_python_semantic/src/semantic_index/builder.rs#L320-L335

Reply:
Ohh, this is the semantic builder and not the type inference builder. But I'm glad it helped to discover another usage :laughing: